std.debug: colour improvements

adtac opened the pull request:
Previously, ZIG_DEBUG_COLOR's presence or absence was used. Instead,
parsing the actual value (if present) allows turning off colour in stack
traces, for example.

adtac commented:
This is my first PR so please let me know if I could do anything better.

mikdusan commented:
on IRC it sounds like the way forward is to support `--color` command line option with values `always` or `never` and `auto` or anything else meaning auto.

and we'd like that to be the same for `ZIG_DEBUG_COLOR`.

please adjust `wantTtyColor()` to return true when  ZIG_DEBUG_COLOR=always, false when ZIG_DEBUG_COLOR=never, otherwise default to automatic behaviour.

mikdusan commented:
also doc/docgen.zig uses `ZIG_DEBUG_COLOR` and should be updated to the new standard

adtac commented:
Ah, I thought we wanted auto/always/never for just `--color`. Also, I wanted to maintain `ZIG_COLOR_DEBUG` compatibility with the only documented use in `docgen.zig`.

Anyway, updated. Rebased on master.

adtac commented on std/debug.zig:
```diff
@@ -73,7 +73,14 @@ pub fn getSelfDebugInfo() !*DebugInfo {
+            return true;
+        } else if (mem.eql(u8, sval, "never")) {
+            return false;
+        }
```
If there is a more idiomatic way to do this, please let me know.

andrewrk replied:
Looks fine to me!

andrewrk commented on std/debug.zig:
```diff
@@ -736,22 +743,6 @@ fn printLineInfo(
             symbol_name,
             compile_unit_name,
         );
-        if (printLineFromFile(out_stream, line_info)) {
```
It was actually intentional that disabling "color" also disabled printing source lines. The idea is that the "color" parameter is choosing between 2 modes:
 * machine readable mode, which does not use color and does not print the line from the file or the caret
 * human readable mode, which uses color and prints the line from the file and the caret

adtac replied:
Have them as two separate options? When the output is not a TTY, use terse stack traces. But if the output is a terminal, but the user dislikes colours, give them an option to disable just that.

justinian replied:
Pardon my jumping in to add my $0.02, but doesn't this implicit behavior mean there's no way to pipe such debug output to a pager (or to `vim -` as I often do) without the color code noise? It seems like there should be separate explicit switches for this.

andrewrk replied:
These are reasonable proposals to make; let's discuss in a separate issue.

andrewrk commented:
Please feel free to re-open this with the requested changes.
